src/checkout/lib/index.ts

```diff
@@ -27,19 +27,19 @@ export async function createPaymentSession(
 }
 
 export async function fetchPaymentSession(
-  sessionId: string,
-  campaignId: string
+  campaignId: string,
+  sessionId: string | null
 ): Promise<void> {
-  const res = await fetch(
-    buildAdServerEndpoint(
-      `/ads/checkout-session?sessionId=${sessionId}&referenceId=${campaignId}`
-    ),
-    {
-      method: "PUT",
-      mode: "cors",
-      credentials: "include",
-    }
-  );
+  let baseUrl = `/ads/checkout-session?referenceId=${campaignId}`;
```

Comment:
Going forward, making this use `URLSearchParams` with `.set` is probably clearer than string concatenation.